util/src/arbitrum.rs

```diff
@@ -31,8 +32,42 @@ pub fn parse_addr_from_deployments_file(file_path: &str, contract_key: &str) ->
         .ok_or_else(|| eyre!("Could not parse {contract_key} address from deployments file"))
 }
 
-/// Get a copy of the protocol's encryption key
-pub fn get_protocol_encryption_key() -> EncryptionKey {
-    jubjub_from_hex_string(PROTOCOL_ENCRYPTION_KEY)
-        .expect("contract encryption key is not a valid jubjub point")
+/// Get the protocol fee from the static variable
+///
+/// Panics if the protocol fee has not been set
+pub fn get_protocol_fee() -> FixedPoint {
+    let fee = PROTOCOL_FEE.get();
+
+    // If the mocks feature is enabled we unwrap to a default
+    #[cfg(feature = "mocks")]
+    {
+        *fee.unwrap_or(&FixedPoint::from_f64_round_down(0.0006)) // 6 bps
+    }
+
+    #[cfg(not(feature = "mocks"))]
+    {
+        *fee.expect("Protocol fee not set")
+    }
+}
+
+/// Get the protocol encryption key from the static variable
+///
+/// Panics if the protocol encryption key has not been set
+pub fn get_protocol_pubkey() -> EncryptionKey {
+    let key = PROTOCOL_PUBKEY.get();
+
+    // If the mocks feature is enabled we unwrap to a default
+    #[cfg(feature = "mocks")]
+    {
+        use circuit_types::elgamal::DecryptionKey;
+        use rand::thread_rng;
+
+        let mut rng = thread_rng();
+        *key.unwrap_or(&DecryptionKey::random(&mut rng).public_key())
```

Comment:
should we `.set` the protocol pubkey so it's a constant pubkey when it's mocked?

Reply:
good suggestion